internal/controller/user_controller.go

```diff
@@ -132,8 +132,8 @@ func (wr *wrappedUserReconciler) Reconcile(ctx context.Context, mdbClient *sqlCl
 		if err := mdbClient.CreateUser(ctx, accountName, createUserOpts...); err != nil {
 			return fmt.Errorf("error creating User: %v", err)
 		}
-	} else if password != "" {
-		if err := mdbClient.AlterUser(ctx, username, password); err != nil {
+	} else if password != "" || wr.user.Spec.MaxUserConnections > 0 {
+		if err := mdbClient.AlterUser(ctx, username, password, wr.user.Spec.MaxUserConnections); err != nil {
```

Comment:
What if the password is non empty but the connections are a zero value? My suggestion is that `AlterUser` takes this into account and avoids altering the `MAX_USER_CONNECTIONS` whenever the maxconnections param is 0.

Reply:
Understood!! :D , I've configured `AlterUser` to take this into account and changed above code accordingly. 
```
else if password != "" || wr.user.Spec.MaxUserConnections > 0 {
		if err := mdbClient.AlterUser(ctx, username, sqlClient.WithIdentifiedBy(password), sqlClient.WithMaxUserConnections(wr.user.Spec.MaxUserConnections)); err != nil {
			return fmt.Errorf("error altering User: %v", err)
		}
	}
```

Please let me know if there is any change needed. :)